usr/share/rear/lib/layout-functions.sh

```diff
@@ -752,10 +752,10 @@ function get_part_device_name_format() {
 function apply_layout_mappings() {
 
     # apply_layout_mappings need one argument (a non-empty file which contains disk device to migrate).
-    if [ -s "$1" ] ; then
-        file_to_migrate="$1"
+    if [ -z $1 ] ; then
```

Comment:
This must be quoted...

Reply:
This one was not quoted because I don't want a single space to be a valid argument.

As @jsmeix  said:
```
A=" " 

[ -z "$A" ] && echo "empty" || echo "not empty"
not empty

[ -z $A ] && echo "empty" || echo "not empty"
empty
```

Reply:
Sorry, that is not the most relevant test. What you should look at is this code example:

```bash
$ foo=" " ; [ $foo ] && echo yes || echo no
no
$ foo="hello world" ; [ $foo ] && echo yes || echo no
bash: [: hello: unary operator expected
no
```

As you can see in the case of an argument with a blank the test fails even though it should succeed. And there is no check that can distinguish between the unary operator error and the content of the test, Bash as a programming language is not that sophisticated.

Therefore IMHO we should ignore the case of an argument consisting of `" "` (unless you can show me a likely case for that to happen) and harden the code against unexpected input. In general, the input to a program should never cause unhandled errors and this unary operator error is exactly an unhandled error.

If you - for whatever reason - really want to exclude an empty string as a parameter then please make an extra test for it, e.g. `[ "$1" = " " ]`. However, in this case I don't see the value of that test as you will have to check for the parameter to be a file in the filesystem. The blank string won't pass that test and nothing bad will happen.

Reply:
@schabrolles 
you may add an explanatory comment
why it is intentionally not quoted in this case
to avoid that another contributor may "just fix" it ;-)

Reply:
@schlomo Ok, It's clear. thanks